spec.bs

```diff
@@ -130,110 +129,65 @@ are:
-contributions cache=] given the worklet's [=contributions cache=] and the
-worklet's global scope.
-
-Issue: Does Protected Audience API have one global scope per auction or
-    multiple? If multiple, will need to change scope for batching.
-
-Issue: How to handle fenced frame-triggered contributions and other
-    event-triggered contributions.
-
-Issue: Need to handle `auctionReportBuyers` and `auctionReportBuyerKeys` here or
-    in the Protected Audience API spec.
-
-<div algorithm>
-The <dfn method for="PrivateAggregation">contributeToHistogramOnEvent(DOMString
-event, PAExtendedHistogramContribution contribution)</dfn> method steps are:
-</div>
 
-Issue: Fill in validation. Will need to check offsets are valid and in the right
-    range for the type etc. Also that base values are one of a set.
+Exposing to global scopes {#exposing}
+=====================================
 
-Issue: Need to document limits on offset, etc.
+To expose this API to a global scope, a readonly attribute of type
+{{PrivateAggregation}} named `privateAggregation` should be exposed on the
+global scope.
 
-Issue: Fill in the rest. (Need to put the contribution in some sort of queue and
-    process the queue at some point. Need to decide where the queue should live
-    given it has to outlive the auction.)
-
-To <dfn>process the Protected Audience contributions cache</dfn> given a
-[=list=] |contributionsCache| and a {{InterestGroupScriptRunnerGlobalScope}}
-|scope|, perform the following steps:
-1. Let |filledInContributions| be a new [=list/is empty|empty=] [=list=].
-1. [=list/iterate|For each=] |contribution| of |contributionsCache|:
-    1. [=list/Append=] the result of [=filling in the contribution=] given
-        |contribution| to |filledInContributions|.
-1. Perform the [=PrivateAggregation/report creation and scheduling steps=] with
-    |scope|'s [=relevant settings object=]'s [=environment settings object/
-    origin=], "<code>[=context type/fledge=]</code>" and
-    |filledInContributions|.
+Additionally, each global scope should set the [=PrivateAggregation/get batching
+scope steps=] for the {{PrivateAggregation}} object it exposes to a non-null
+value. The global scope may wait to set the field to indicate that the API is
+not yet available. For any batching scope returned by the [=PrivateAggregation/
+get batching scope steps=], the [=mark a batching scope complete=] steps should
+later be performed given that same batching scope, the global scope's [=relevant
+settings object=]'s [=environment settings object/origin=] and some [=context
+type=].
 
+Note: This last requirement means that global scopes with different origins
+    cannot share the same batching scope.
 
 Structures {#structures}
 ========================
 
 General {#general-structures}
 -----------------------------
 
-<h4 dfn-type=dfn>Aggregatable report</h3>
+<h4 dfn-type=dfn>Batching scope</h4>
+A batching scope is a unique internal value that identifies which
```

Comment:
s/unique internal value/implementation-defined value/

https://infra.spec.whatwg.org/#implementation-defined

Reply:
For this, I was trying to emulate the "opaque origin" kind of concept. I saw "unique internal value" in the HTML spec (https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#unique-values), but I don't think it's exported, so I might need to move that definition in here.

Reply:
@domenic - thoughts? Should we try to export "unique internal value"?

Reply:
Huh I didn't realize that we made that into a whole concept with a definition. Probably we should move that to Infra and make it exported.

In the meantime, you could do `<a spec=HTML>unique internal value</a>` to link to the concept despite it not being exported.

Reply:
Thanks, moved to the <a spec=HTML> tag as suggested. Also added an issue and filed https://github.com/whatwg/infra/issues/583 to track this.